script/prep_data.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+#import torch
+#import torch.nn as nn
+#import torch.nn.functional as F
+#import torch.optim as optim
+from torch.utils.data import Dataset #, DataLoader
+
+class MNIST_Dataset(Dataset):
+    def __init__(self, image, binary = True):
+        ###### why no self here?
+        ###### why use super?
+        super(MNIST_Dataset).__init__()
+        self.image = image
+        # is binary labels? 
+        self.binary = binary
+        # len and get item used by DataLoader function
+    def __len__(self):
+        return self.image.shape[0]
+    def __getitem__(self, idx):
+        # turn each row into 28x28 image
+        image = np.copy(self.image[idx, :].reshape(28,28))
+        
+        if self.binary:
+            
+            ### binarize MNIST images
+            ####### not binarizing??? 
+            # Otsu algorithm: 
+            # iteratively searches for the threshold that minimizes 
+            # the within-class variance, defined as a weighted sum of 
+            # variances of the two classes (background and foreground)
+            
+            nbins = 0.01
+            all_colors = image.flatten()
+            total_weight = len(all_colors)
+            least_variance = -1
+            least_variance_threshold = -1
+            
+            # create an array of all possible threshold values which we want to loop through
+            color_thresholds = np.arange(np.min(image)+nbins, np.max(image)-nbins, nbins)
+            
+            # loop through the thresholds to find the one with the least within class variance
+            for color_threshold in color_thresholds:
```

Comment:
IIRC, you modified this block of code from the original TD-VAE code, correct?

If so, please add a comment to this effect

Reply:
added a comment about changing that portion of code